src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -245,37 +235,90 @@ public void smokeTest() throws Exception {
     }
 
     @Test public void shebang() throws Exception {
+        setGlobalInterpreter("/bin/false"); // Should be overridden
         Controller c = new BourneShellScript("#!/bin/cat\nHello, world!").launch(new EnvVars(), ws, launcher, listener);
-        while (c.exitStatus(ws, launcher, listener) == null) {
-            Thread.sleep(100);
-        }
+        awaitCompletion(c);
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         c.writeLog(ws, new TeeOutputStream(baos, System.out));
         assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
         assertThat(baos.toString(), containsString("Hello, world!"));
         c.cleanup(ws);
     }
 
+    @Issue("JENKINS-50902")
+    @Test public void configuredInterpreter() throws Exception {
+        // Run script inside the container as this is system dependent
+        DumbSlave node = createDockerSlave(dockerSlim.get());
+        j.jenkins.addNode(node);
+        node.toComputer().waitUntilOnline();
```

Comment:
nit: I think `JenkinsRule#waitOnline(Slave)` would return better error messages in the case of failure, but the behavior should be the same, so not important.

Reply:
Good point, addressed.